### PR TITLE
ceph: fix to support TLS enabled rgw-endpoint

### PR DESCRIPTION
**Pull request by aruniiird**
Signed-off-by: Arun Kumar Mohan <amohan@redhat.com>

**Description of your changes:**
Enables the 'create-external-cluster-resources.py' script to check a given TLS (https) enabled rgw-endpoint.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py [hunk 1]
```diff
@@ -116,16 +116,17 @@ def _invalid_endpoint(self, endpoint_str):
...
-                raise ExecutionFailureException(
-                    "wrong return code {} on rgw endpoint http header request".format(rc))
-        except requests.ConnectionError:
-            raise ExecutionFailureException(
-                "failed to connect to rgw endpoint {}".format(ep))
+        protocols = ["http", "https"]
+        for prefix in protocols:
+            try:
+                ep = "{}://{}".format(prefix, endpoint_str)
+                r = requests.head(ep)
```
Comment (leseb) [inline cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:123]: What's the timeout all ready for this? Can we shorten it to a few seconds? Like <5s?

Comment (aruniiird) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py]: I have added 3 second as per the following [doc](https://github.com/psf/requests/blob/master/requests/api.py),
```
:param timeout: (optional) How many seconds to wait for the server to send data
        before giving up, as a float, or a :ref:`(connect timeout, read timeout) <timeouts>` tuple.
        :type timeout: float or tuple
```

Comment (leseb) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py]: Did you test your change?

**Review by leseb — APPROVED** [review]

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py (hunk 1 above)
Comment (aruniiird) [reply cluster/examples/kubernetes/ceph/create-external-cluster-resources.py]: Yes I have tested my change individually (by providing URLs like golang , youtube websites and other non-existing) on a separate python file. It works within the timeout.

```python
import sys
import requests

class ExecutionFailureException(Exception):
    pass

def endpoint_check(endpoint_str, timeout=5):
    protocols = ["http", "https"]
    for prefix in protocols:
        try:
            ep = "{}://{}".format(prefix, endpoint_str)
            print("EP: {}".format(ep))
            r = requests.head(ep, timeout=timeout)
            if r.status_code == 200:
                return
            print("EP Status: {}".format(r.status_code))
        except:
            continue
    raise ExecutionFailureException(
        "unable to connect to endpoint: {}".format(endpoint_str))


def main():
    endpoint = "vlang.io"
    if len(sys.argv) > 1:
        endpoint = sys.argv[1]
    endpoint_check(endpoint, timeout=3)

if __name__ == "__main__":
    main()
```

**Pull request merged**